test/integration.sh

```diff
@@ -1,47 +1,89 @@
+            else
+                counter=$((counter+1))
+            fi
+        done
 
-    INFO "doing upload with $(pwd)/manifest.tsv"
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/manifest.tsv:/dcc/manifest.tsv -v $(pwd)/samples:/samples -v $(pwd)/outputs:/outputs \
-         quay.io/ucsc_cgl/core-client:1.1.0-alpha spinnaker-upload --skip-submit --force-upload /dcc/manifest.tsv
+        if [[ "$counter" -eq $num_containers_exp ]]; then
+            ERROR "One or more docker containers not running."\
+		&& exit 1
+        fi
+    done
 
-    object_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 20)
-    bundle_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 19)
-    INFO "the uploaded bundle with id ${bundle_id} has metadata.json with object id ${object_id}"
+}
+
+function main {
+    # This tests whether all Docker containers are installed and then
+    # hits the DCC Dashboard Host and checks whether it returns a 200
+    # status code.
+
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    LOG_LEVEL_ALL
+    check_setup
 
-    INFO "downloading metadata.json with object id ${object_id}"
-    sudo mkdir outputs/test_download
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/outputs:/outputs quay.io/ucsc_cgl/core-client:1.1.0-alpha download ${object_id} /outputs/test_download
+    # Get name of the DCC Dashboard Host.
+    curr_dir=$(pwd)
+    cd ../boardwalk
+    dcc_host=$(grep -o 'dcc_dashboard_service.*' .env | cut -f2- -d=)
+    # Check from environment variable value whether user is white-listed.    
+    whitelist_set=$(grep -o 'email_white.*' .env | cut -f2- -d=)
+    cd "$curr_dir"
 
-    [[ ! -f outputs/test_download/${bundle_id}/metadata.json ]] && printf "%s: no such file" "outputs/test_download/${bundle_id}/metadata.json" | FATAL && exit 1
+    # Check if home page is returned. If so it will not output anything,
+    # otherwise it aborts and prints cURL error code to stdout.
+    curl -fsS "https://$dcc_host" > /dev/null
 
-    sudo rm -rf outputs
+    # Get status code of response to export functionality.
+    url="https://$dcc_host/api/v1/repository/files/export"
+    response=$(curl -sIL "$url")  # no output, just headers, follow redirects
+    status_code=$(echo "$response" | grep 'HTTP.*' | cut -f2- -d " ")
 
-    # TODO: run indexer, etc.
+    # If the whitelist string is empty user isn't white-listed, and cURL
+    # is expected to return 200; if user is white-listed a 401 is the expected
+    # response.
+    if [[ ! "$status_code" == *"401"* ]] && [[ ! -z "$whitelist_set" ]]; then
+        ERROR "Did not get expected status code 401 from host $dcc_host" && exit 1
+    elif [[ ! "$status_code" == *"200"* ]]\
+	     && [[ ! "$status_code" == *"401"* ]]; then
+	ERROR "Did not get expected status code 200 from host $dcc_host" && exit 1
```

Comment:
The message doesn't reflect the condition on lines 81 and 82.

Reply:
Fixed it by returning HTTP status code in that case.